CMakeLists.txt

```diff
@@ -1,5 +1,12 @@
 cmake_minimum_required(VERSION 3.5)
 
+# Set policies up to 3.9 since we want to enable the IPO option
+if(${CMAKE_VERSION} VERSION_LESS 3.9)
```

Comment:
i don't understand, the goal is to set cmae_policy to `at least` 3.9 no ? then why setup policies to less than 3.9 on versions before 3.9 ?
We should also may bump the cmake_minimum_required, as 3.5 is very very old

Reply:
Older versions cannot set policies that only exist in newer versions; you will get an error if you try. The point of this is to make sure that whatever CMake version you're using, whatever policies it does have will be enabled, and it should probably set policies all the way up to 3.21, but I didn't want to be hasty to do that without reviewing what those policies are and what unexpected effect it might have.